frontend/src/visualization/filtering/Filter.tsx

```diff
@@ -64,140 +70,75 @@ function TabPanel(props: TabPanelProps) {
   );
 }
 
-/**
- * Fetches nodeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
- */
-function fetchNodeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<NodeType[]> {
-  return schemaService.getNodeTypes(cancellation);
-}
-
-/**
- * Fetches edgeTypes from the schemaService.
- *
- * @param schemaService - the schemaService the data is fetched from
- * @param cancellation - the cancellation token
- */
-function fetchEdgeTypes(
-  schemaService: SchemaService,
-  cancellation: CancellationToken
-): Promise<EdgeType[]> {
-  return schemaService.getEdgeTypes(cancellation);
-}
-
 const Filter = (): JSX.Element => {
   // hooks
   const classes = useStyles();
   const [tabIndex, setTabIndex] = React.useState(0);
   const [open, setOpen] = React.useState(false);
 
-  const schemaService = useService(SchemaService, null);
-
-  const filterStore = useService(FilterQueryStore);
-
-  const nodeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-  const edgeConditionsRef = useRef<(FilterCondition | null)[]>([]);
-
-  function updateQuery() {
-    const nodeConditions = nodeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
-
-    const edgeConditions = edgeConditionsRef.current.filter(
-      (condition) => condition !== null
-    ) as FilterCondition[];
+  const filterStateStore = useService<FilterStateStore>(FilterStateStore);
 
-    const filters: { nodes?: FilterCondition; edges?: FilterCondition } = {};
-
-    if (nodeConditions.length > 0) {
-      filters.nodes = MatchAnyCondition(...nodeConditions);
-    }
-
-    if (edgeConditions.length > 0) {
-      filters.edges = MatchAnyCondition(...edgeConditions);
-    }
-
-    // TODO: Make limits configurable
-    filterStore.mergeState({ filters });
-  }
-
-  // a JSX.Element template used for rendering
-  const entityTemplate = (
-    color: string,
-    name: string,
-    entity: 'node' | 'edge',
-    i: number
-  ) => {
-    const setEntryFilterCondition = (
-      condition: FilterCondition | null
-    ): void => {
-      const conditionsRef =
-        entity === 'node' ? nodeConditionsRef : edgeConditionsRef;
-      const conditions = conditionsRef.current;
+  const schemaService = useService(SchemaService, null);
```

Comment:
No need to provide null as second parameter.
```suggestion
  const schemaService = useService(SchemaService);
```